models/comment.js

```diff
@@ -89,6 +92,15 @@ const CommentSchema = new Schema({
   }
 });
 
+// Add the indexes on the comment tag.
+CommentSchema.index({
+  'id': 1,
+  'tags.id': 1
```

Comment:
The tags id should not be a part of this index, it is then coupling the id of the comment with the id of the tag, which doesn't make sense.